pkg/response/formatter.go

```diff
@@ -0,0 +1,211 @@
+type OutputFormat int
+
+const (
+	// JSON displays the output in JSON format
+	JSON OutputFormat = iota
+	// HUMAN displays the outut in a way that's nice for humans to read
+	HUMAN
+	// TOML displays the output in TOML format
+	TOML
+	// YAML displays the output in YAML format
+	YAML
+	// CSV displays the
+	CSV
+)
+
+type Formatter struct {
+	format OutputFormat
+}
+
+// NewFormatter creates new formatter
+func NewFormatter(cfg config.Formatter) (*Formatter, error) {
+	format, err := GetOutputFormat(cfg.Format)
+	if err != nil {
+		return nil, err
+	}
+	return &Formatter{format: format}, nil
+}
+
+// GetOutputFormat takes the string and returns OutputFormat or an error
+func GetOutputFormat(format string) (OutputFormat, error) {
+	format = strings.ToUpper(format)
+	switch format {
+	case "JSON":
+		return JSON, nil
+	case "HUMAN":
+		return HUMAN, nil
+	case "TOML":
+		return TOML, nil
+	case "YAML":
+		return YAML, nil
+	case "CSV":
+		return CSV, nil
+	default:
+		return JSON, fmt.Errorf("invalid output format option: format=%q", format)
+	}
+}
+
+// SetFormat sets the format using the OutputFormat value
+func (f *Formatter) SetFormat(format OutputFormat) {
+	f.format = format
+}
+
+// Format renders a response structure to the set format as a string
+func (f *Formatter) Format(r *Response) string {
+	var output string
+	switch f.format {
+	case JSON:
+		output = formatJson(r)
+	case HUMAN:
+		output = formatHuman(r)
+	case TOML:
+		output = formatToml(r)
+	case YAML:
+		output = formatYaml(r)
+	case CSV:
+		output = formatCsv(r)
+	}
+	return output
+}
+
+func formatJson(r *Response) string {
+	out, err := json.Marshal(r)
+	if err != nil {
+		logger.Error("could not marshal response: error=%q", err)
+	}
+	return string(out)
+}
+
+func formatHuman(r *Response) string {
+	headers := flattenedResponseFields()
+	for i, header := range headers {
+		// Append a : to each label. Do it once here instead of every loop
+		headers[i] = header + ":"
+	}
+	flat := flattenedResponse(r)
+	var out []string
+	for _, response := range flat {
+		for i, entry := range response {
+			// Specifies width of 26 characters for labels
+			out = append(out, fmt.Sprintf("%-26s%s", headers[i], entry))
+		}
+		out = append(out, "\n")
+	}
+	return strings.Join(out, "\n")
+}
+
+func formatToml(r *Response) string {
+	var buf bytes.Buffer
+
+	if err := toml.NewEncoder(&buf).Encode(r); err != nil {
+		logger.Error("could not marshal response: error=%q", err)
+	}
+	return buf.String()
+}
+
+func formatYaml(r *Response) string {
+	out, err := yaml.Marshal(r)
+	if err != nil {
+		logger.Error("could not marshal response: error=%q", err)
+	}
+	return string(out)
+}
+
+func formatCsv(r *Response) string {
+	headers := flattenedResponseFields()
+
+	var buf bytes.Buffer
+	writer := csv.NewWriter(&buf)
+
+	err := writer.Write(headers)
+	if err != nil {
+		logger.Error("could not write response: error=%q", err)
+	}
+
+	err = writer.WriteAll(flattenedResponse(r))
```

Comment:
Thoughts on only doing the sanizieField on the CSV one?

```suggestion
        	for _, row := range flattenedResponse(r) {
		sanitizedRow := make([]string, len(row))
		for i, field := range row {
			sanitizedRow[i] = sanitizeField(field)
		}

		err = writer.Write(sanitizedRow)
		if err != nil {
			break
		}
	}

```

Rationale: 

For the human readable one, we'd want to show exactly what was matched for the sake of someone grepping for it.

Only issue for the human one I see is for multi-line values. But for gitleaks they handle it just by truncating things and letting it wrap:

```
Finding:     -----BEGIN PRIVATE KEY-----                                     
MIIEvQIBADANBgkqhkiG9w0BAQEFAASCBKcwggSjAgEAAoIBAQDW8OVrrcJ/rZxZ
/kHlgz5...                                                      
Secret:      -----BEGIN PRIVATE KEY-----                                     
MIIEvQIBADANBgkqhkiG9w0BAQEFAASCBKcwggSjAgEAAoIBAQDW8OVrrcJ/rZxZ
/kHlgz5...                                                      
RuleID:      private-key
Entropy:     6.028506
File:        pki/valid/private.key
Line:        2
Fingerprint: pki/valid/private.key:private-key:2
```

Reply:
Now only sanatize on CSV. 

There is also a truncate option for truncating commonly long lines.